src/wallet/wallet.h

```diff
@@ -1286,8 +1285,7 @@ class WalletRescanReserver
     bool reserve()
     {
         assert(!m_could_reserve);
-        std::lock_guard<std::mutex> lock(m_wallet.mutexScanning);
```

Comment:
960d64d326c501dbe22b80094171d7fe3e86bd69
It seems this changes behavior as the whole block is guarded by mutex currently.

Reply:
Same reasoning as previous; except that this one sets `fScanningWallet` to `true` twice :(

Reply:
So why not dropping the following `m_wallet.fScanningWallet = true;` ?

Reply:
As `WalletRescanReserver.m_wallet` is a reference, the object being referenced to could be shared among different threads. So not using the `mutexScanning` mutex here is not thread-safe, IMO.

Reply:
dropped redundant `fScanningWallet = true`